File: extract_derexXD_comments.py
import subprocess
import json
from yt_dlp import YoutubeDL

flags = {
    "skip_download": True,
    "writesubtitles": False,
    "writeautomaticsub": False,
    "getcomments": True,  
    }

# hard coded lol
url = 'https://youtu.be/ytKVGtJ5yng?si=2bq3ltobiuo9lA4q'

def merge(arr, left, mid, right):
    """I wrote a merge sort, it is 5:32 AM and Texas Winter Storm apparently about to hit"""

def mergeSort(list, left, right):
    """Split everything into tiny pieces. Then we begin sorting as the pieces are compared and put back together"""
    if left < right:
        mid = (left + right) // 2
        mergeSort(list, left, mid)
        mergeSort(list, mid + 1, right)
        merge(list, left, mid, right)



with YoutubeDL(flags) as derexXD:
    info = derexXD.extract_info(url, download=False)

comments = info.get("comments", [])

with open("comments.json", "w") as file:
    json.dump(comments, file, indent=2)

print(f"Done\nExtracted {len(comments)} comments from el video")

# subprocess.run(
#     [
#         "yt-dlp",
#         "--skip-download",
#         "--write-comments",
#         "--no-write-info-json",
#         "--output",
#         "comments",
#         url,
#     ],
#     check=True
# )

File: extract_derexXD_comments.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
	"sort"
)

// hard coded lol
const url = "https://youtu.be/ytKVGtJ5yng?si=2bq3ltobiuo9lA4q"

func main() {
	cmd := exec.Command(
		"yt-dlp",
		"--skip-download",
		"--write-comments",
		"--dump-json",
		"--no-progress",
		"--no-warnings",
		url,
	)
	out, err := cmd.Output()
	if err != nil {
		if ee, ok := err.(*exec.ExitError); ok {
			fmt.Fprintf(os.Stderr, "yt-dlp failed:\n%s\n", ee.Stderr)
		}
		panic(err)
	}

	var info struct {
		Comments []map[string]any `json:"comments"`
	}
	if err := json.Unmarshal(out, &info); err != nil {
		panic(err)
	}

	comments := info.Comments
	if comments == nil {
		comments = []map[string]any{}
	}

	sort.SliceStable(comments, func(i, j int) bool {
		return likeCount(comments[i]) > likeCount(comments[j])
	})

	data, err := json.MarshalIndent(comments, "", "  ")
	if err != nil {
		panic(err)
	}
	data = append(data, '\n')

	outPath := filepath.Join("docs", "comments.json")
	if err := os.WriteFile(outPath, data, 0644); err != nil {
		panic(err)
	}

	fmt.Printf("Done\nExtracted %d comments from el video\n", len(comments))
}

func likeCount(comment map[string]any) float64 {
	n, _ := comment["like_count"].(float64)
	return n
}
